File: core/src/com/nopalsoft/thetruecolor/leaderboard/DialogRanking.java
package com.nopalsoft.thetruecolor.leaderboard;

import com.badlogic.gdx.Application.ApplicationType;
import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.graphics.Color;
import com.badlogic.gdx.scenes.scene2d.Group;
import com.badlogic.gdx.scenes.scene2d.InputEvent;
import com.badlogic.gdx.scenes.scene2d.ui.Button;
import com.badlogic.gdx.scenes.scene2d.ui.Image;
import com.badlogic.gdx.scenes.scene2d.ui.Label;
import com.badlogic.gdx.scenes.scene2d.ui.ScrollPane;
import com.badlogic.gdx.scenes.scene2d.ui.Table;
import com.badlogic.gdx.scenes.scene2d.utils.ClickListener;
import com.badlogic.gdx.scenes.scene2d.utils.NinePatchDrawable;
import com.badlogic.gdx.scenes.scene2d.utils.TextureRegionDrawable;
import com.nopalsoft.thetruecolor.Assets;
import com.nopalsoft.thetruecolor.screens.MainMenuScreen;
import com.nopalsoft.thetruecolor.screens.BaseScreen;
import com.nopalsoft.thetruecolor.scene2d.DialogFacebook;
import com.nopalsoft.thetruecolor.scene2d.DialogAmazon;
import com.nopalsoft.thetruecolor.scene2d.DialogGoogle;
import com.nopalsoft.thetruecolor.TrueColorGame;



public class DialogRanking extends Group {
    public static final float WIDTH = 400;
    public static final float HEIGHT = 385;

    MainMenuScreen menuScreen;
    TrueColorGame game;

    Label rankingTitle;

    Button btFacebook;
    Button btGoogle;

    DialogFacebook dialogFacebook;
    DialogGoogle dialogGoogle;
    DialogAmazon dialogAmazon;

    Table container;

    public DialogRanking(MainMenuScreen screen) {
        menuScreen = screen;
        game = screen.game;
        setBounds(BaseScreen.SCREEN_WIDTH / 2f - WIDTH / 2f, 210, WIDTH, HEIGHT);
        setBackground(Assets.dialogRanking);

        rankingTitle = new Label(Assets.languages.get("ranking"), new Label.LabelStyle(Assets.fontSmall, Color.WHITE));
        rankingTitle.setPosition(15, 328);

        dialogFacebook = new DialogFacebook(screen);
        dialogGoogle = new DialogGoogle(screen);
        dialogAmazon = new DialogAmazon(screen);

        btFacebook = new Button(Assets.buttonFacebook);

        menuScreen.addPressEffect(btFacebook);
        btFacebook.addListener(new ClickListener() {
            @Override
            public void clicked(InputEvent event, float x, float y) {
                dialogFacebook.show(getStage());
            }
        });

        TextureRegionDrawable btLoginKeyFrame = Assets.buttonGoogle;


        btGoogle = new Button(btLoginKeyFrame);

        menuScreen.addPressEffect(btGoogle);
        btGoogle.addListener(new ClickListener() {
        });

        Table tableSocial = new Table();
        tableSocial.setSize(130, 50);
        tableSocial.setPosition(255, 328);
        tableSocial.defaults().expandX().size(50).right();
        tableSocial.add(btFacebook);

        if (Gdx.app.getType() != ApplicationType.WebGL && Gdx.app.getType() != ApplicationType.iOS) {
            tableSocial.add(btGoogle);
        }

        addActor(rankingTitle);
        addActor(tableSocial);

        container = new Table();

        ScrollPane scroll = new ScrollPane(container);
        scroll.setSize(WIDTH, 320);
        scroll.setPosition(0, 0);

        addActor(scroll);

        container.top();
    }

    private void setBackground(NinePatchDrawable dialogRanking) {
        Image imageDialogRanking = new Image(dialogRanking);
        imageDialogRanking.setSize(getWidth(), getHeight());
        addActor(imageDialogRanking);

    }

    public void addPerson(Person person) {
        container.add(person);
        container.row();
    }

    public void clearLeaderboard() {
        container.clear();
    }
}

File: core/src/com/nopalsoft/thetruecolor/leaderboard/DialogRanking.kt
package com.nopalsoft.thetruecolor.leaderboard

import com.badlogic.gdx.Application.ApplicationType
import com.badlogic.gdx.Gdx
import com.badlogic.gdx.graphics.Color
import com.badlogic.gdx.scenes.scene2d.Group
import com.badlogic.gdx.scenes.scene2d.InputEvent
import com.badlogic.gdx.scenes.scene2d.ui.Button
import com.badlogic.gdx.scenes.scene2d.ui.Image
import com.badlogic.gdx.scenes.scene2d.ui.Label
import com.badlogic.gdx.scenes.scene2d.ui.ScrollPane
import com.badlogic.gdx.scenes.scene2d.ui.Table
import com.badlogic.gdx.scenes.scene2d.utils.ClickListener
import com.badlogic.gdx.scenes.scene2d.utils.NinePatchDrawable
import com.nopalsoft.thetruecolor.Assets
import com.nopalsoft.thetruecolor.scene2d.DialogFacebook
import com.nopalsoft.thetruecolor.screens.BaseScreen
import com.nopalsoft.thetruecolor.screens.MainMenuScreen

class DialogRanking(screen: MainMenuScreen) : Group() {


    private val rankingTitle = Label(Assets.languages["ranking"], Label.LabelStyle(Assets.fontSmall, Color.WHITE))
    private val btFacebook = Button(Assets.buttonFacebook)
    private val btLoginKeyFrame = Assets.buttonGoogle
    private val btGoogle = Button(btLoginKeyFrame)
    private val dialogFacebook = DialogFacebook(screen)
    private val container = Table()


    init {
        setBounds(BaseScreen.SCREEN_WIDTH / 2f - WIDTH / 2f, 210f, WIDTH, HEIGHT)
        setBackground(Assets.dialogRanking)
        rankingTitle.setPosition(15f, 328f)

        screen.addPressEffect(btFacebook)
        btFacebook.addListener(object : ClickListener() {
            override fun clicked(event: InputEvent, x: Float, y: Float) {
                dialogFacebook.show(stage)
            }
        })

        screen.addPressEffect(btGoogle)
        btGoogle.addListener(object : ClickListener() {
        })

        val tableSocial = Table()
        tableSocial.setSize(130f, 50f)
        tableSocial.setPosition(255f, 328f)
        tableSocial.defaults().expandX().size(50f).right()
        tableSocial.add(btFacebook)

        if (Gdx.app.type != ApplicationType.WebGL && Gdx.app.type != ApplicationType.iOS) {
            tableSocial.add(btGoogle)
        }

        addActor(rankingTitle)
        addActor(tableSocial)

        val scroll = ScrollPane(container)
        scroll.setSize(WIDTH, 320f)
        scroll.setPosition(0f, 0f)

        addActor(scroll)
        container.top()
    }

    private fun setBackground(dialogRanking: NinePatchDrawable) {
        val imageDialogRanking = Image(dialogRanking)
        imageDialogRanking.setSize(width, height)
        addActor(imageDialogRanking)
    }

    fun addPerson(person: Person) {
        container.add(person)
        container.row()
    }

    fun clearLeaderboard() {
        container.clear()
    }

    companion object {

        const val WIDTH = 400F
        private const val HEIGHT = 385F
    }
}